packages/engine-server/src/drivers/SQLiteMetadataStore.ts

```diff
@@ -0,0 +1,171 @@
+import {
+  asyncLoopOneAtATime,
+  DVault,
+  NoteProps,
+  NotePropsByIdDict,
+  VaultUtils,
+} from "@dendronhq/common-all";
+import fs from "fs-extra";
+import _ from "lodash";
+import { PrismaClient } from "../generated-prisma-client";
+
+let _prisma: PrismaClient | undefined;
+
+function getPrismaClient(): PrismaClient {
+  if (_prisma === undefined) {
+    throw Error("prisma client not initialized");
+  }
+  return _prisma;
+}
+
+export type NoteIndexLightProps = {
+  fname: string;
+  id: string;
+  foo: string;
+};
+
+export class SQLiteMetadataStore {
```

Comment:
this should implement `IDataStore<string, NotePropsMeta>` to make sure it's compatible with new engine architecture.